packages/rocketchat-ui/client/views/app/room.js

```diff
@@ -229,43 +229,7 @@ Template.room.helpers({
 	},
 
 	messagesHistory() {
-		const hideMessagesOfType = [];
-		settings.collection.find({ _id: /Message_HideType_.+/ }).forEach(function(record) {
-			let types;
-			const type = record._id.replace('Message_HideType_', '');
-			switch (type) {
-				case 'mute_unmute':
-					types = ['user-muted', 'user-unmuted'];
-					break;
-				default:
-					types = [type];
-			}
-			return types.forEach(function(type) {
-				const index = hideMessagesOfType.indexOf(type);
-
-				if ((record.value === true) && (index === -1)) {
-					hideMessagesOfType.push(type);
-				} else if (index > -1) {
-					hideMessagesOfType.splice(index, 1);
-				}
-			});
-		});
-
-		const query =
-			{ rid: this._id };
-
-		if (hideMessagesOfType.length > 0) {
-			query.t =
-				{ $nin: hideMessagesOfType };
-		}
-
-		const options = {
-			sort: {
-				ts: 1,
-			},
-		};
-
-		return ChatMessage.find(query, options);
+		return Template.instance().messages.get();
```

Comment:
I made some tests here, and that helper just runs 1 time, is the cursor "ChatMessage.find(query, options)" slower than the other approach?

Reply:
It is cuz it returns a cursor that triggers the render for every new record while here it's returning a reactive var that receives all data in a single operation.